From fdd27bab5f134fda0098f9c3a5c36199e3205fea Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Thu, 4 Jul 2024 18:04:18 +0200
Subject: [PATCH 20/21] gdbserver_tests filters remove more verbose python rpm
 load warnings

The rpm python load warnings on Fedora became even more verbose.
Filter out all stderr messages related to them so tests don't fail.

(cherry picked from commit d73a8e745bf09cfc2d44e14f2ddd70fae2893d1d)
---
 gdbserver_tests/filter_gdb.in  | 5 ++++-
 gdbserver_tests/filter_vgdb.in | 7 +++++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/gdbserver_tests/filter_gdb.in b/gdbserver_tests/filter_gdb.in
index f0400fd18be6..38a3803eb1f8 100755
--- a/gdbserver_tests/filter_gdb.in
+++ b/gdbserver_tests/filter_gdb.in
@@ -52,7 +52,10 @@ s/^\[?1034hReading symbols/Reading symbols/
 /^Try: zypper install -C/d
 
 #      Remove python rpm module load warning
-/^Unable to load 'rpm' module.  Please install the python3-rpm package./d
+/^Unable to load /d
+/^the RPM suggestions /d
+/^installing missing debuginfo /d
+/^please install /d
 
 #       Remove vgdb message
 /relaying data between gdb and process/d
diff --git a/gdbserver_tests/filter_vgdb.in b/gdbserver_tests/filter_vgdb.in
index 71f1f702537e..b32be689733b 100755
--- a/gdbserver_tests/filter_vgdb.in
+++ b/gdbserver_tests/filter_vgdb.in
@@ -47,8 +47,11 @@ $SED -e '/Use.*info auto-load python-scripts.*/d' |
 $SED -e '/warning: Invalid entry in .debug_gdb_scripts section/d' |
 $SED -e "/Use .info auto-load python .REGEXP.. to list them./d" |
 
-# Remove python rpm module load warning
-$SED -e "/^Unable to load 'rpm' module.  Please install the python3-rpm package./d" |
+# Remove python rpm module load warnings
+$SED -e "/^Unable to load /d" |
+$SED -e "/^the RPM suggestions /d" |
+$SED -e "/^installing missing debuginfo /d" |
+$SED -e "/^please install /d" |
 
 # and filter out any remaining empty lines
 $SED -e '/^$/d'
-- 
2.45.2

